exonum-java-binding-core/src/main/java/com/exonum/binding/service/adapters/UserServiceAdapter.java

```diff
@@ -151,19 +157,28 @@ public void mountPublicApiHandler(long nodeNativeHandle) {
   public void afterCommit(long snapshotHandle, int validatorId, long height) {
     assert snapshotHandle != 0;
 
+    BlockCommittedEvent event = null;
     try (Cleaner cleaner = new Cleaner("UserServiceAdapter#afterCommit")) {
       Snapshot snapshot = viewFactory.createSnapshot(snapshotHandle, cleaner);
       OptionalInt optionalValidatorId = validatorId >= 0
           ? OptionalInt.of(validatorId)
           : OptionalInt.empty();
-      BlockCommittedEvent event =
-          BlockCommittedEventImpl.valueOf(snapshot, optionalValidatorId, height);
-      service.afterCommit(event);
+      event = BlockCommittedEventImpl.valueOf(snapshot, optionalValidatorId, height);
+      doAfterCommit(event);
     } catch (CloseFailuresException e) {
+      LOGGER.error("Failed to close resources during after commit event {}", event, e);
       throw new RuntimeException(e);
     }
   }
 
+  private void doAfterCommit(BlockCommittedEvent event) {
+    try {
+      service.afterCommit(event);
+    } catch (Exception e) {
+      LOGGER.error("Unexpected exception during after commit event {}", event, e);
```

Comment:
Rather, "An exception in after commit handler" (in user code, not in ours). Also, shan't we reduce to WARN?